modules/tinymce/src/plugins/table/test/ts/browser/command/ModifyClassesCommandsTest.ts

```diff
@@ -0,0 +1,249 @@
+          );
+        };
+
+        const getContentWithClass = () => {
+          return (
+            '<table>' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td class="a">1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        it('Can be toggled on', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithoutClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableCellToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithClass());
+        });
+
+        it('Can be toggled off', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableCellToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithoutClass());
+        });
+      });
+
+      context('When there is already a class on the cell, which should not be deleted', () => {
+        const getContentWithoutClass = () => {
+          return (
+            '<table>' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td class="b">1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        const getContentWithClass = () => {
+          return (
+            '<table>' +
+              '<tbody>' +
+                '<tr>' +
+                  '<td class="b a">1</td>' +
+                  '<td>2</td>' +
+                '</tr>' +
+              '</tbody>' +
+            '</table>'
+          );
+        };
+
+        it('Can be toggled on', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithoutClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableCellToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithClass());
+        });
+
+        it('Can be toggled off', () => {
+          const editor = hook.editor();
+          editor.setContent(getContentWithClass());
+
+          TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1, true);
+
+          execCmdAndAssertEvent('TINY-7163', editor, 'mceTableCellToggleClass', false, 'a');
+
+          assert.equal(cleanTableHtml(editor.getContent()), getContentWithoutClass());
+        });
+      });
+    });
+
+    context('Can use mceTableToggleClass to toggle the table class', () => {
+      context('When there is no class already on the cell', () => {
```

Comment:
This description doesn't match, it's not doing operations on the cell and instead is doing it on the table.